admin/class-admin-apple-meta-boxes.php

```diff
@@ -386,26 +305,84 @@ public static function section_is_checked( $sections, $section_id, $is_default )
 	/**
 	 * Registers assets used by meta boxes.
 	 *
-	 * @param string $hook
+	 * @param string $hook The initiator of the action hook.
+	 *
 	 * @access public
 	 */
 	public function register_assets( $hook ) {
+
+		// Only fire on post and new post views.
 		if ( 'post.php' !== $hook && 'post-new.php' !== $hook ) {
 			return;
 		}
 
-		wp_enqueue_style( $this->plugin_slug . '_meta_boxes_css', plugin_dir_url(
-			__FILE__ ) .  '../assets/css/meta-boxes.css' );
+		// Ensure media modal assets are enqueued.
+		wp_enqueue_media();
 
-		wp_enqueue_script( $this->plugin_slug . '_meta_boxes_js', plugin_dir_url(
-			__FILE__ ) .  '../assets/js/meta-boxes.js', array( 'jquery' ),
-			self::$version, true );
+		// Enqueue metabox stylesheet.
+		wp_enqueue_style(
+			$this->plugin_slug . '_meta_boxes_css',
+			plugin_dir_url( __FILE__ ) .  '../assets/css/meta-boxes.css'
+		);
+
+		// Enqueue cover art stylesheet.
+		wp_enqueue_style(
+			$this->plugin_slug . '_cover_art_css',
+			plugin_dir_url( __FILE__ ) .  '../assets/css/cover-art.css'
+		);
+
+		// Enqueue metabox script.
+		wp_enqueue_script(
+			$this->plugin_slug . '_meta_boxes_js',
+			plugin_dir_url( __FILE__ ) .  '../assets/js/meta-boxes.js',
+			array( 'jquery' ),
+			self::$version,
+			true
+		);
+
+		// Enqueue cover art script.
+		wp_enqueue_script(
+			$this->plugin_slug . '_cover_art_js',
+			plugin_dir_url( __FILE__ ) .  '../assets/js/cover-art.js',
+			array( 'jquery' ),
+			self::$version,
+			true
+		);
 
-		// Localize the JS file for handling frontend actions.
+		// Localize the JS file for meta boxes.
 		wp_localize_script( $this->plugin_slug . '_meta_boxes_js', 'apple_news_meta_boxes', array(
 			'publish_action' => $this->publish_action,
 		) );
 
+		// Localize the JS file for cover art.
+		wp_localize_script( $this->plugin_slug . '_cover_art_js', 'apple_news_cover_art', array(
+			'image_sizes' => Admin_Apple_News::$image_sizes,
+			'image_too_small' => esc_html__( 'You must select an image that is at least the minimum height and width specified above.', 'apple-news' ),
+			'media_modal_button' => esc_html__( 'Select image', 'apple-news' ),
+			'media_modal_title' => esc_html__( 'Choose an image', 'apple-news' ),
+		) );
 	}
 
+	/**
+	 * Saves a cover art image to meta, given a post ID and a field name.
+	 *
+	 * @param int $post_id The post ID to update meta for.
+	 * @param string $field_name The field name to use in POST and meta_key.
+	 *
+	 * @access private
+	 */
+	private static function _save_coverart_meta( $post_id, $field_name ) {
+
+		// Determine if there is postdata for this key.
+		if ( empty( $_POST[ $field_name ] )
+			|| ! is_numeric( $_POST[ $field_name ] )
```

Comment:
Even though you check `is_numeric` VIP will still likely complain if you don't sanitize below via `absint` or whatever is most appropriate.